<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8" />
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />

<meta name="viewport" content="width=device-width, initial-scale=1">

<meta name="author" content="Beibei Jiang" />

<meta name="date" content="2018-09-13" />

<title>SimPhe</title>



<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
a.sourceLine { display: inline-block; line-height: 1.25; }
a.sourceLine { pointer-events: none; color: inherit; text-decoration: inherit; }
a.sourceLine:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode { white-space: pre; position: relative; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
code.sourceCode { white-space: pre-wrap; }
a.sourceLine { text-indent: -1em; padding-left: 1em; }
}
pre.numberSource a.sourceLine
  { position: relative; left: -4em; }
pre.numberSource a.sourceLine::before
  { content: attr(data-line-number);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; pointer-events: all; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {  }
@media screen {
a.sourceLine::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>



<style type="text/css">body {
background-color: #fff;
margin: 1em auto;
max-width: 700px;
overflow: visible;
padding-left: 2em;
padding-right: 2em;
font-family: "Open Sans", "Helvetica Neue", Helvetica, Arial, sans-serif;
font-size: 14px;
line-height: 1.35;
}
#header {
text-align: center;
}
#TOC {
clear: both;
margin: 0 0 10px 10px;
padding: 4px;
width: 400px;
border: 1px solid #CCCCCC;
border-radius: 5px;
background-color: #f6f6f6;
font-size: 13px;
line-height: 1.3;
}
#TOC .toctitle {
font-weight: bold;
font-size: 15px;
margin-left: 5px;
}
#TOC ul {
padding-left: 40px;
margin-left: -1.5em;
margin-top: 5px;
margin-bottom: 5px;
}
#TOC ul ul {
margin-left: -2em;
}
#TOC li {
line-height: 16px;
}
table {
margin: 1em auto;
border-width: 1px;
border-color: #DDDDDD;
border-style: outset;
border-collapse: collapse;
}
table th {
border-width: 2px;
padding: 5px;
border-style: inset;
}
table td {
border-width: 1px;
border-style: inset;
line-height: 18px;
padding: 5px 5px;
}
table, table th, table td {
border-left-style: none;
border-right-style: none;
}
table thead, table tr.even {
background-color: #f7f7f7;
}
p {
margin: 0.5em 0;
}
blockquote {
background-color: #f6f6f6;
padding: 0.25em 0.75em;
}
hr {
border-style: solid;
border: none;
border-top: 1px solid #777;
margin: 28px 0;
}
dl {
margin-left: 0;
}
dl dd {
margin-bottom: 13px;
margin-left: 13px;
}
dl dt {
font-weight: bold;
}
ul {
margin-top: 0;
}
ul li {
list-style: circle outside;
}
ul ul {
margin-bottom: 0;
}
pre, code {
background-color: #f7f7f7;
border-radius: 3px;
color: #333;
white-space: pre-wrap; 
}
pre {
border-radius: 3px;
margin: 5px 0px 10px 0px;
padding: 10px;
}
pre:not([class]) {
background-color: #f7f7f7;
}
code {
font-family: Consolas, Monaco, 'Courier New', monospace;
font-size: 85%;
}
p > code, li > code {
padding: 2px 0px;
}
div.figure {
text-align: center;
}
img {
background-color: #FFFFFF;
padding: 2px;
border: 1px solid #DDDDDD;
border-radius: 3px;
border: 1px solid #CCCCCC;
margin: 0 5px;
}
h1 {
margin-top: 0;
font-size: 35px;
line-height: 40px;
}
h2 {
border-bottom: 4px solid #f7f7f7;
padding-top: 10px;
padding-bottom: 2px;
font-size: 145%;
}
h3 {
border-bottom: 2px solid #f7f7f7;
padding-top: 10px;
font-size: 120%;
}
h4 {
border-bottom: 1px solid #f7f7f7;
margin-left: 8px;
font-size: 105%;
}
h5, h6 {
border-bottom: 1px solid #ccc;
font-size: 105%;
}
a {
color: #0033dd;
text-decoration: none;
}
a:hover {
color: #6666ff; }
a:visited {
color: #800080; }
a:visited:hover {
color: #BB00BB; }
a[href^="http:"] {
text-decoration: underline; }
a[href^="https:"] {
text-decoration: underline; }

code > span.kw { color: #555; font-weight: bold; } 
code > span.dt { color: #902000; } 
code > span.dv { color: #40a070; } 
code > span.bn { color: #d14; } 
code > span.fl { color: #d14; } 
code > span.ch { color: #d14; } 
code > span.st { color: #d14; } 
code > span.co { color: #888888; font-style: italic; } 
code > span.ot { color: #007020; } 
code > span.al { color: #ff0000; font-weight: bold; } 
code > span.fu { color: #900; font-weight: bold; }  code > span.er { color: #a61717; background-color: #e3d2d2; } 
</style>

</head>

<body>




<h1 class="title toc-ignore">SimPhe</h1>
<h4 class="author"><em>Beibei Jiang</em></h4>
<h4 class="date"><em>2018-09-13</em></h4>



<p>This vignette introduces how to use <em>SimPhe</em>, an R package dedicated to simulate multiple phenotyes based on genotyping data. The main feature of this package is the possibility to take genetic epistatic effects, not only additive-additive interaction but also additive-dominance and dominance-dominance, heritability and multiple phenotypes into account. Moreover, we provide a variety of convenient functions, including suggestion for set the variation of random variable according to user specified genetic effects and flexible support for input formats, as well as output formats. It also supports the input of plink formats.</p>
<div id="installation" class="section level2">
<h2>Installation</h2>
<div class="sourceCode" id="cb1"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb1-1" data-line-number="1"><span class="kw">install.packages</span>(SimPhe)</a></code></pre></div>
</div>
<div id="sample-implementation" class="section level2">
<h2>Sample implementation</h2>
<div class="sourceCode" id="cb2"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb2-1" data-line-number="1"><span class="co"># load package</span></a>
<a class="sourceLine" id="cb2-2" data-line-number="2"><span class="kw">library</span>(SimPhe)</a></code></pre></div>
<p>First, we show how easy to get phenotype(s) by using <em>sim.phe</em>. Before running <em>sim.phe</em>, we need specify the parameter file and genotype file for simulation. After installing <em>SimPhe</em>, these two toy files exist in package folder. To get the path, type</p>
<div class="sourceCode" id="cb3"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb3-1" data-line-number="1"><span class="co"># get file path of simulation parameters</span></a>
<a class="sourceLine" id="cb3-2" data-line-number="2"><span class="co"># (two shared SNP pairs and one independent SNP pair for each phenotype)</span></a>
<a class="sourceLine" id="cb3-3" data-line-number="3">fpar.path &lt;-<span class="st"> </span><span class="kw">system.file</span>(<span class="st">&quot;extdata&quot;</span>, <span class="st">&quot;simupars.txt&quot;</span>, <span class="dt">package=</span><span class="st">&quot;SimPhe&quot;</span>)</a>
<a class="sourceLine" id="cb3-4" data-line-number="4"></a>
<a class="sourceLine" id="cb3-5" data-line-number="5"><span class="co"># get file path of genotype file: rows are individuals and columns are SNPs</span></a>
<a class="sourceLine" id="cb3-6" data-line-number="6">fgeno.path &lt;-<span class="st"> </span><span class="kw">system.file</span>(<span class="st">&quot;extdata&quot;</span>, <span class="st">&quot;10SNP.txt&quot;</span>, <span class="dt">package=</span><span class="st">&quot;SimPhe&quot;</span>)</a></code></pre></div>
<p>Then simulate the phenotypes as designed in the parameter file after loading package:</p>
<div class="sourceCode" id="cb4"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb4-1" data-line-number="1"><span class="co"># simulate phenotype(s)</span></a>
<a class="sourceLine" id="cb4-2" data-line-number="2">phe &lt;-<span class="st"> </span><span class="kw">sim.phe</span>(<span class="dt">sim.pars =</span> fpar.path, </a>
<a class="sourceLine" id="cb4-3" data-line-number="3">               <span class="dt">fgeno =</span> fgeno.path, </a>
<a class="sourceLine" id="cb4-4" data-line-number="4">               <span class="dt">ftype =</span> <span class="st">&quot;snp.head&quot;</span>, </a>
<a class="sourceLine" id="cb4-5" data-line-number="5">               <span class="dt">seed =</span> <span class="dv">123</span>, </a>
<a class="sourceLine" id="cb4-6" data-line-number="6">               <span class="dt">fwrite =</span> <span class="ot">FALSE</span>)</a></code></pre></div>
<p>In the parameter file, we conduct two phenotypes contrbituted by two common SNP pairs with epistatic effects and one independent SNP pair with epistatic effects, the example of simulation parameters, same as the example file, is also included in <em>SimPhe</em>. Users could touch it by typing <em>genepars</em>.</p>
<div class="sourceCode" id="cb5"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb5-1" data-line-number="1">genepars</a></code></pre></div>
<pre><code>## $P1mean
##   mean
## 1   10
## 
## $P1main
##     SNP additive dominance
## 1 SNP01      4.0       2.0
## 2 SNP02      2.5       1.5
## 3 SNP03      5.5       1.8
## 4 SNP04     -2.5       1.0
## 5 SNP05      8.5       9.8
## 6 SNP06      8.7       8.4
## 
## $P1epistasis
##    SNPA  SNPB additive_additive additive_dominance dominance_additive
## 1 SNP01 SNP02              10.0                2.0                2.8
## 2 SNP03 SNP04               8.0                3.0                3.5
## 3 SNP05 SNP06              20.5               20.6               20.4
##   dominance_dominance
## 1                 1.8
## 2                 2.5
## 3                10.2
## 
## $P1heritability
##   heritability
## 1          0.6
## 
## $P2mean
##   mean
## 1   20
## 
## $P2main
##     SNP additive dominance
## 1 SNP01      8.0       2.0
## 2 SNP02      4.5       1.8
## 3 SNP03      6.5       1.5
## 4 SNP04     -3.5      -1.2
## 5 SNP07     10.4       8.6
## 6 SNP08     12.3      11.9
## 
## $P2epistasis
##    SNPA  SNPB additive_additive additive_dominance dominance_additive
## 1 SNP01 SNP02              15.0                4.0                3.0
## 2 SNP03 SNP04              12.0                5.0                3.5
## 3 SNP07 SNP08              21.5               20.9               20.5
##   dominance_dominance
## 1                 2.0
## 2                 2.5
## 3                20.3</code></pre>
<p>Phenotype 1 has been set with certain heritability but phenotype 2 has not. We will show whether the heritability of the simulated phenotype 1 is same as the set. <em>SimPhe</em> includes the coefficients and the allele frequencies for simulating phenotype 1: <em>gene.coef</em> and <em>allele.freq</em>, which is extracted from the simulation parameters.</p>
<div class="sourceCode" id="cb7"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb7-1" data-line-number="1"><span class="co"># regression coefficients for simulating phenotype 1</span></a>
<a class="sourceLine" id="cb7-2" data-line-number="2">gene.coefficients</a></code></pre></div>
<pre><code>## $epi.par1
##    SNPA  SNPB additiveA dominanceA additiveB dominanceB additive_additive
## 1 SNP01 SNP02         4          2       2.5        1.5                10
##   additive_dominance dominance_additive dominance_dominance
## 1                  2                2.8                 1.8
## 
## $epi.par2
##    SNPA  SNPB additiveA dominanceA additiveB dominanceB additive_additive
## 1 SNP03 SNP04       5.5        1.8      -2.5          1                 8
##   additive_dominance dominance_additive dominance_dominance
## 1                  3                3.5                 2.5
## 
## $epi.par3
##    SNPA  SNPB additiveA dominanceA additiveB dominanceB additive_additive
## 1 SNP05 SNP06       8.5        9.8       8.7        8.4              20.5
##   additive_dominance dominance_additive dominance_dominance
## 1               20.6               20.4                10.2</code></pre>
<div class="sourceCode" id="cb9"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb9-1" data-line-number="1"><span class="co"># allele frequencies of SNPs for simulating phenotype 1</span></a>
<a class="sourceLine" id="cb9-2" data-line-number="2">allele.freq</a></code></pre></div>
<pre><code>##     SNP major.frequency minor.frequency
## 1 SNP05       0.6447673       0.3552327
## 2 SNP01       0.7238366       0.2761634
## 3 SNP03       0.7394188       0.2605812
## 4 SNP06       0.7334176       0.2665824
## 5 SNP02       0.6433986       0.3566014
## 6 SNP04       0.6581386       0.3418614</code></pre>
<div class="sourceCode" id="cb11"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb11-1" data-line-number="1"><span class="co"># calculate genetic variance</span></a>
<a class="sourceLine" id="cb11-2" data-line-number="2">genevar &lt;-<span class="st"> </span><span class="kw">calc.gene.var</span>(gene.coefficients, allele.freq)</a>
<a class="sourceLine" id="cb11-3" data-line-number="3"></a>
<a class="sourceLine" id="cb11-4" data-line-number="4"><span class="co"># the variance of simulated phenotype 1</span></a>
<a class="sourceLine" id="cb11-5" data-line-number="5">phe1var &lt;-<span class="st"> </span><span class="kw">var</span>(phe[, <span class="st">&quot;p1&quot;</span>])</a>
<a class="sourceLine" id="cb11-6" data-line-number="6"></a>
<a class="sourceLine" id="cb11-7" data-line-number="7"><span class="co"># heritability of simulated phenotype 1</span></a>
<a class="sourceLine" id="cb11-8" data-line-number="8">simuht &lt;-<span class="st"> </span>genevar <span class="op">/</span><span class="st"> </span>phe1var</a>
<a class="sourceLine" id="cb11-9" data-line-number="9">simuht</a></code></pre></div>
<pre><code>## [1] 0.6297038</code></pre>
<p>The result is not the exactly 0.6 due to the (pseudo)random numbers generated in R. To get phenotype 2 with a specific heritability, for example, 0.45, we could proceed as:</p>
<div class="sourceCode" id="cb13"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb13-1" data-line-number="1"><span class="co"># get regression coefficients settings for phenotype 2</span></a>
<a class="sourceLine" id="cb13-2" data-line-number="2">genecoef &lt;-<span class="st"> </span><span class="kw">get.gene.coef</span>(<span class="dt">main.pars =</span> <span class="kw">specify.pars</span>(<span class="dt">genetic.pars =</span> genepars,</a>
<a class="sourceLine" id="cb13-3" data-line-number="3">                                                   <span class="dt">effect.type  =</span> <span class="st">&quot;main&quot;</span>,</a>
<a class="sourceLine" id="cb13-4" data-line-number="4">                                                   <span class="dt">phe.index    =</span> <span class="dv">2</span>),</a>
<a class="sourceLine" id="cb13-5" data-line-number="5">                          <span class="dt">epi.pars  =</span> <span class="kw">specify.pars</span>(<span class="dt">genetic.pars =</span> genepars,</a>
<a class="sourceLine" id="cb13-6" data-line-number="6">                                                   <span class="dt">effect.type  =</span> <span class="st">&quot;epistasis&quot;</span>,</a>
<a class="sourceLine" id="cb13-7" data-line-number="7">                                                   <span class="dt">phe.index    =</span> <span class="dv">2</span>))</a>
<a class="sourceLine" id="cb13-8" data-line-number="8"><span class="co"># get genotype data</span></a>
<a class="sourceLine" id="cb13-9" data-line-number="9">genotype &lt;-<span class="st"> </span><span class="kw">read.geno</span>(<span class="dt">fname =</span> fgeno.path, <span class="dt">ftype =</span> <span class="st">&quot;snp.head&quot;</span>)</a>
<a class="sourceLine" id="cb13-10" data-line-number="10"></a>
<a class="sourceLine" id="cb13-11" data-line-number="11"><span class="co"># get allele frequencies of the SNPs set for phenotype 2</span></a>
<a class="sourceLine" id="cb13-12" data-line-number="12">freq2 &lt;-<span class="st"> </span><span class="kw">get.freq</span>(<span class="dt">geno =</span> genotype,</a>
<a class="sourceLine" id="cb13-13" data-line-number="13">                  <span class="dt">epi.pars  =</span> <span class="kw">specify.pars</span>(<span class="dt">genetic.pars =</span> genepars,</a>
<a class="sourceLine" id="cb13-14" data-line-number="14">                                           <span class="dt">effect.type  =</span> <span class="st">&quot;epistasis&quot;</span>,</a>
<a class="sourceLine" id="cb13-15" data-line-number="15">                                           <span class="dt">phe.index    =</span> <span class="dv">2</span>))</a>
<a class="sourceLine" id="cb13-16" data-line-number="16"></a>
<a class="sourceLine" id="cb13-17" data-line-number="17"><span class="co"># get noise variation</span></a>
<a class="sourceLine" id="cb13-18" data-line-number="18">exp.noise.var &lt;-<span class="st"> </span><span class="kw">get.noise.var</span>(<span class="dt">gene.coef =</span> genecoef,</a>
<a class="sourceLine" id="cb13-19" data-line-number="19">                               <span class="dt">freq =</span> freq2,</a>
<a class="sourceLine" id="cb13-20" data-line-number="20">                               <span class="dt">heritability =</span> <span class="fl">0.45</span>)</a></code></pre></div>
<p>Then when simulating a phenotype, just give this value as argument <em>noise.var</em> to function <em>sim.phe</em>. It will generate a phenotype which has a heritability close to 0.45.</p>
<p>As we mentioned earlier in this article, buiding the correlation by setting the shared interactive SNP pairs cannot be controlled. We can take a look at the correlation between the simulated phenotype 1 and phenotype 2:</p>
<div class="sourceCode" id="cb14"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb14-1" data-line-number="1"><span class="co"># correlation test</span></a>
<a class="sourceLine" id="cb14-2" data-line-number="2"><span class="kw">cor.test</span>(phe[, <span class="st">&quot;p1&quot;</span>], phe[, <span class="st">&quot;p2&quot;</span>])</a></code></pre></div>
<pre><code>## 
##  Pearson's product-moment correlation
## 
## data:  phe[, &quot;p1&quot;] and phe[, &quot;p2&quot;]
## t = 2.2112, df = 98, p-value = 0.02935
## alternative hypothesis: true correlation is not equal to 0
## 95 percent confidence interval:
##  0.0225377 0.3973936
## sample estimates:
##       cor 
## 0.2179908</code></pre>
<p>According to the result of correlation test, the two simulated phenotypes are significantly correlated but the correlation coefficient is small and the amount cannot easily be predicted. To get a certain amount or higher correlation, we can conduct correlation by applying the correlation matrix to two independent variables. For two phenotypes, if we set different SNP pairs for each, we assume these two phenotypes are independent. Here we give another parameter file to but same genotype file to the arguments in <em>sim.phe</em>, <em>fgenetic.pars</em> and <em>fgeno</em>:</p>
<div class="sourceCode" id="cb16"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb16-1" data-line-number="1"><span class="co"># get file path of simulation parameters</span></a>
<a class="sourceLine" id="cb16-2" data-line-number="2"><span class="co"># (different SNP pairs for each phenotype)</span></a>
<a class="sourceLine" id="cb16-3" data-line-number="3">fpar.path &lt;-<span class="st"> </span><span class="kw">system.file</span>(<span class="st">&quot;extdata&quot;</span>, <span class="st">&quot;sep_simupars.txt&quot;</span>, <span class="dt">package=</span><span class="st">&quot;SimPhe&quot;</span>)</a>
<a class="sourceLine" id="cb16-4" data-line-number="4"></a>
<a class="sourceLine" id="cb16-5" data-line-number="5"><span class="co"># simulate phenotype(s)</span></a>
<a class="sourceLine" id="cb16-6" data-line-number="6">indphe &lt;-<span class="st"> </span><span class="kw">sim.phe</span>(<span class="dt">sim.pars =</span> fpar.path,</a>
<a class="sourceLine" id="cb16-7" data-line-number="7">                  <span class="dt">fgeno =</span> fgeno.path,</a>
<a class="sourceLine" id="cb16-8" data-line-number="8">                  <span class="dt">ftype =</span> <span class="st">&quot;snp.head&quot;</span>,</a>
<a class="sourceLine" id="cb16-9" data-line-number="9">                  <span class="dt">seed =</span> <span class="dv">123</span>,</a>
<a class="sourceLine" id="cb16-10" data-line-number="10">                  <span class="dt">fwrite =</span> <span class="ot">FALSE</span>)</a></code></pre></div>
<p>We can test the correlation between initial phenotypes with seperated SNP pairs settings:</p>
<div class="sourceCode" id="cb17"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb17-1" data-line-number="1"><span class="co"># test original correlation</span></a>
<a class="sourceLine" id="cb17-2" data-line-number="2"><span class="kw">cor.test</span>(indphe[, <span class="st">&quot;p1&quot;</span>], indphe[, <span class="st">&quot;p2&quot;</span>])</a></code></pre></div>
<pre><code>## 
##  Pearson's product-moment correlation
## 
## data:  indphe[, &quot;p1&quot;] and indphe[, &quot;p2&quot;]
## t = -0.094853, df = 98, p-value = 0.9246
## alternative hypothesis: true correlation is not equal to 0
## 95 percent confidence interval:
##  -0.2056123  0.1871892
## sample estimates:
##          cor 
## -0.009581143</code></pre>
<p>Apparently, these two phenotypes are not related. We could continue our work on coverting them to be correlated. First, conduct a correlation matrix:</p>
<div class="sourceCode" id="cb19"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb19-1" data-line-number="1"><span class="co"># correlation matrix</span></a>
<a class="sourceLine" id="cb19-2" data-line-number="2">corm &lt;-<span class="st"> </span><span class="kw">matrix</span>(<span class="kw">c</span>(<span class="dv">1</span>, <span class="fl">0.6</span>, <span class="fl">0.6</span>, <span class="dv">1</span>), <span class="dt">ncol =</span> <span class="dv">2</span>)</a>
<a class="sourceLine" id="cb19-3" data-line-number="3">corm</a></code></pre></div>
<pre><code>##      [,1] [,2]
## [1,]  1.0  0.6
## [2,]  0.6  1.0</code></pre>
<p>Before applying correlation matrix to simulated phenotypes, we would like to know what the data looks like:</p>
<div class="sourceCode" id="cb21"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb21-1" data-line-number="1"><span class="kw">apply</span>(indphe, <span class="dv">2</span>, mean)</a></code></pre></div>
<pre><code>##       p1       p2 
## 20.18028 33.24045</code></pre>
<div class="sourceCode" id="cb23"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb23-1" data-line-number="1"><span class="kw">apply</span>(indphe, <span class="dv">2</span>, sd)</a></code></pre></div>
<pre><code>##       p1       p2 
## 20.05408 17.92715</code></pre>
<p>Then we can build correlation between the two initial phenotypes:</p>
<div class="sourceCode" id="cb25"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb25-1" data-line-number="1"><span class="co"># build correlation</span></a>
<a class="sourceLine" id="cb25-2" data-line-number="2">corphe &lt;-<span class="st"> </span><span class="kw">build.cor.phe</span>(indphe, <span class="dt">corMtr =</span> corm)</a></code></pre></div>
<p>Now let’s test the correlation between the two new phenotypes and see if there is anything difference:</p>
<div class="sourceCode" id="cb26"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb26-1" data-line-number="1"><span class="co"># check mean and standard deviation of new data set</span></a>
<a class="sourceLine" id="cb26-2" data-line-number="2"><span class="kw">apply</span>(corphe, <span class="dv">2</span>, mean)</a></code></pre></div>
<pre><code>##       p1       p2 
## 20.18028 33.24045</code></pre>
<div class="sourceCode" id="cb28"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb28-1" data-line-number="1"><span class="kw">apply</span>(corphe, <span class="dv">2</span>, sd)</a></code></pre></div>
<pre><code>##       p1       p2 
## 20.05408 17.84452</code></pre>
<div class="sourceCode" id="cb30"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb30-1" data-line-number="1"><span class="co"># test correlation</span></a>
<a class="sourceLine" id="cb30-2" data-line-number="2"><span class="kw">cor.test</span>(corphe[, <span class="st">&quot;p1&quot;</span>], corphe[, <span class="st">&quot;p2&quot;</span>])</a></code></pre></div>
<pre><code>## 
##  Pearson's product-moment correlation
## 
## data:  corphe[, &quot;p1&quot;] and corphe[, &quot;p2&quot;]
## t = 7.3301, df = 98, p-value = 6.615e-11
## alternative hypothesis: true correlation is not equal to 0
## 95 percent confidence interval:
##  0.4514244 0.7086646
## sample estimates:
##       cor 
## 0.5950781</code></pre>
<p>Obviously, there is no significant difference on means and standard deviations between the initial phenotypes and the new phenotypes.</p>
</div>
<div id="references" class="section level1 unnumbered">
<h1>References</h1>
<div id="refs" class="references">
<div id="ref-cockerham1977quadratic">
<p>Cockerham, C. Clark, and Bruce Spencer Weir. 1977. “Quadratic Analyses of Reciprocal Crosses.” <em>Biometrics</em> 33 (1). JSTOR: 187–203. <a href="https://doi.org/10.2307/2529312" class="uri">https://doi.org/10.2307/2529312</a>.</p>
</div>
<div id="ref-kao2002modeling">
<p>Kao, Chen-Hung, and Zhao-Bang Zeng. 2002. “Modeling Epistasis of Quantitative Trait Loci Using Cockerham’s Model.” <em>Genetics</em> 160 (3). Genetics Society of America: 1243–61. <a href="https://doi.org/10.1534/genetics.104.035857" class="uri">https://doi.org/10.1534/genetics.104.035857</a>.</p>
</div>
</div>
</div>



<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
