<!DOCTYPE html>
<html>
<head>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8"/>

<title>Installation</title>

<script type="text/javascript">
window.onload = function() {
  var imgs = document.getElementsByTagName('img'), i, img;
  for (i = 0; i < imgs.length; i++) {
    img = imgs[i];
    // center an image if it is the only element of its parent
    if (img.parentElement.childElementCount === 1)
      img.parentElement.style.textAlign = 'center';
  }
};
</script>

<!-- Styles for R syntax highlighter -->
<style type="text/css">
   pre .operator,
   pre .paren {
     color: rgb(104, 118, 135)
   }

   pre .literal {
     color: #990073
   }

   pre .number {
     color: #099;
   }

   pre .comment {
     color: #998;
     font-style: italic
   }

   pre .keyword {
     color: #900;
     font-weight: bold
   }

   pre .identifier {
     color: rgb(0, 0, 0);
   }

   pre .string {
     color: #d14;
   }
</style>

<!-- R syntax highlighter -->
<script type="text/javascript">
var hljs=new function(){function m(p){return p.replace(/&/gm,"&amp;").replace(/</gm,"&lt;")}function f(r,q,p){return RegExp(q,"m"+(r.cI?"i":"")+(p?"g":""))}function b(r){for(var p=0;p<r.childNodes.length;p++){var q=r.childNodes[p];if(q.nodeName=="CODE"){return q}if(!(q.nodeType==3&&q.nodeValue.match(/\s+/))){break}}}function h(t,s){var p="";for(var r=0;r<t.childNodes.length;r++){if(t.childNodes[r].nodeType==3){var q=t.childNodes[r].nodeValue;if(s){q=q.replace(/\n/g,"")}p+=q}else{if(t.childNodes[r].nodeName=="BR"){p+="\n"}else{p+=h(t.childNodes[r])}}}if(/MSIE [678]/.test(navigator.userAgent)){p=p.replace(/\r/g,"\n")}return p}function a(s){var r=s.className.split(/\s+/);r=r.concat(s.parentNode.className.split(/\s+/));for(var q=0;q<r.length;q++){var p=r[q].replace(/^language-/,"");if(e[p]){return p}}}function c(q){var p=[];(function(s,t){for(var r=0;r<s.childNodes.length;r++){if(s.childNodes[r].nodeType==3){t+=s.childNodes[r].nodeValue.length}else{if(s.childNodes[r].nodeName=="BR"){t+=1}else{if(s.childNodes[r].nodeType==1){p.push({event:"start",offset:t,node:s.childNodes[r]});t=arguments.callee(s.childNodes[r],t);p.push({event:"stop",offset:t,node:s.childNodes[r]})}}}}return t})(q,0);return p}function k(y,w,x){var q=0;var z="";var s=[];function u(){if(y.length&&w.length){if(y[0].offset!=w[0].offset){return(y[0].offset<w[0].offset)?y:w}else{return w[0].event=="start"?y:w}}else{return y.length?y:w}}function t(D){var A="<"+D.nodeName.toLowerCase();for(var B=0;B<D.attributes.length;B++){var C=D.attributes[B];A+=" "+C.nodeName.toLowerCase();if(C.value!==undefined&&C.value!==false&&C.value!==null){A+='="'+m(C.value)+'"'}}return A+">"}while(y.length||w.length){var v=u().splice(0,1)[0];z+=m(x.substr(q,v.offset-q));q=v.offset;if(v.event=="start"){z+=t(v.node);s.push(v.node)}else{if(v.event=="stop"){var p,r=s.length;do{r--;p=s[r];z+=("</"+p.nodeName.toLowerCase()+">")}while(p!=v.node);s.splice(r,1);while(r<s.length){z+=t(s[r]);r++}}}}return z+m(x.substr(q))}function j(){function q(x,y,v){if(x.compiled){return}var u;var s=[];if(x.k){x.lR=f(y,x.l||hljs.IR,true);for(var w in x.k){if(!x.k.hasOwnProperty(w)){continue}if(x.k[w] instanceof Object){u=x.k[w]}else{u=x.k;w="keyword"}for(var r in u){if(!u.hasOwnProperty(r)){continue}x.k[r]=[w,u[r]];s.push(r)}}}if(!v){if(x.bWK){x.b="\\b("+s.join("|")+")\\s"}x.bR=f(y,x.b?x.b:"\\B|\\b");if(!x.e&&!x.eW){x.e="\\B|\\b"}if(x.e){x.eR=f(y,x.e)}}if(x.i){x.iR=f(y,x.i)}if(x.r===undefined){x.r=1}if(!x.c){x.c=[]}x.compiled=true;for(var t=0;t<x.c.length;t++){if(x.c[t]=="self"){x.c[t]=x}q(x.c[t],y,false)}if(x.starts){q(x.starts,y,false)}}for(var p in e){if(!e.hasOwnProperty(p)){continue}q(e[p].dM,e[p],true)}}function d(B,C){if(!j.called){j();j.called=true}function q(r,M){for(var L=0;L<M.c.length;L++){if((M.c[L].bR.exec(r)||[null])[0]==r){return M.c[L]}}}function v(L,r){if(D[L].e&&D[L].eR.test(r)){return 1}if(D[L].eW){var M=v(L-1,r);return M?M+1:0}return 0}function w(r,L){return L.i&&L.iR.test(r)}function K(N,O){var M=[];for(var L=0;L<N.c.length;L++){M.push(N.c[L].b)}var r=D.length-1;do{if(D[r].e){M.push(D[r].e)}r--}while(D[r+1].eW);if(N.i){M.push(N.i)}return f(O,M.join("|"),true)}function p(M,L){var N=D[D.length-1];if(!N.t){N.t=K(N,E)}N.t.lastIndex=L;var r=N.t.exec(M);return r?[M.substr(L,r.index-L),r[0],false]:[M.substr(L),"",true]}function z(N,r){var L=E.cI?r[0].toLowerCase():r[0];var M=N.k[L];if(M&&M instanceof Array){return M}return false}function F(L,P){L=m(L);if(!P.k){return L}var r="";var O=0;P.lR.lastIndex=0;var M=P.lR.exec(L);while(M){r+=L.substr(O,M.index-O);var N=z(P,M);if(N){x+=N[1];r+='<span class="'+N[0]+'">'+M[0]+"</span>"}else{r+=M[0]}O=P.lR.lastIndex;M=P.lR.exec(L)}return r+L.substr(O,L.length-O)}function J(L,M){if(M.sL&&e[M.sL]){var r=d(M.sL,L);x+=r.keyword_count;return r.value}else{return F(L,M)}}function I(M,r){var L=M.cN?'<span class="'+M.cN+'">':"";if(M.rB){y+=L;M.buffer=""}else{if(M.eB){y+=m(r)+L;M.buffer=""}else{y+=L;M.buffer=r}}D.push(M);A+=M.r}function G(N,M,Q){var R=D[D.length-1];if(Q){y+=J(R.buffer+N,R);return false}var P=q(M,R);if(P){y+=J(R.buffer+N,R);I(P,M);return P.rB}var L=v(D.length-1,M);if(L){var O=R.cN?"</span>":"";if(R.rE){y+=J(R.buffer+N,R)+O}else{if(R.eE){y+=J(R.buffer+N,R)+O+m(M)}else{y+=J(R.buffer+N+M,R)+O}}while(L>1){O=D[D.length-2].cN?"</span>":"";y+=O;L--;D.length--}var r=D[D.length-1];D.length--;D[D.length-1].buffer="";if(r.starts){I(r.starts,"")}return R.rE}if(w(M,R)){throw"Illegal"}}var E=e[B];var D=[E.dM];var A=0;var x=0;var y="";try{var s,u=0;E.dM.buffer="";do{s=p(C,u);var t=G(s[0],s[1],s[2]);u+=s[0].length;if(!t){u+=s[1].length}}while(!s[2]);if(D.length>1){throw"Illegal"}return{r:A,keyword_count:x,value:y}}catch(H){if(H=="Illegal"){return{r:0,keyword_count:0,value:m(C)}}else{throw H}}}function g(t){var p={keyword_count:0,r:0,value:m(t)};var r=p;for(var q in e){if(!e.hasOwnProperty(q)){continue}var s=d(q,t);s.language=q;if(s.keyword_count+s.r>r.keyword_count+r.r){r=s}if(s.keyword_count+s.r>p.keyword_count+p.r){r=p;p=s}}if(r.language){p.second_best=r}return p}function i(r,q,p){if(q){r=r.replace(/^((<[^>]+>|\t)+)/gm,function(t,w,v,u){return w.replace(/\t/g,q)})}if(p){r=r.replace(/\n/g,"<br>")}return r}function n(t,w,r){var x=h(t,r);var v=a(t);var y,s;if(v){y=d(v,x)}else{return}var q=c(t);if(q.length){s=document.createElement("pre");s.innerHTML=y.value;y.value=k(q,c(s),x)}y.value=i(y.value,w,r);var u=t.className;if(!u.match("(\\s|^)(language-)?"+v+"(\\s|$)")){u=u?(u+" "+v):v}if(/MSIE [678]/.test(navigator.userAgent)&&t.tagName=="CODE"&&t.parentNode.tagName=="PRE"){s=t.parentNode;var p=document.createElement("div");p.innerHTML="<pre><code>"+y.value+"</code></pre>";t=p.firstChild.firstChild;p.firstChild.cN=s.cN;s.parentNode.replaceChild(p.firstChild,s)}else{t.innerHTML=y.value}t.className=u;t.result={language:v,kw:y.keyword_count,re:y.r};if(y.second_best){t.second_best={language:y.second_best.language,kw:y.second_best.keyword_count,re:y.second_best.r}}}function o(){if(o.called){return}o.called=true;var r=document.getElementsByTagName("pre");for(var p=0;p<r.length;p++){var q=b(r[p]);if(q){n(q,hljs.tabReplace)}}}function l(){if(window.addEventListener){window.addEventListener("DOMContentLoaded",o,false);window.addEventListener("load",o,false)}else{if(window.attachEvent){window.attachEvent("onload",o)}else{window.onload=o}}}var e={};this.LANGUAGES=e;this.highlight=d;this.highlightAuto=g;this.fixMarkup=i;this.highlightBlock=n;this.initHighlighting=o;this.initHighlightingOnLoad=l;this.IR="[a-zA-Z][a-zA-Z0-9_]*";this.UIR="[a-zA-Z_][a-zA-Z0-9_]*";this.NR="\\b\\d+(\\.\\d+)?";this.CNR="\\b(0[xX][a-fA-F0-9]+|(\\d+(\\.\\d*)?|\\.\\d+)([eE][-+]?\\d+)?)";this.BNR="\\b(0b[01]+)";this.RSR="!|!=|!==|%|%=|&|&&|&=|\\*|\\*=|\\+|\\+=|,|\\.|-|-=|/|/=|:|;|<|<<|<<=|<=|=|==|===|>|>=|>>|>>=|>>>|>>>=|\\?|\\[|\\{|\\(|\\^|\\^=|\\||\\|=|\\|\\||~";this.ER="(?![\\s\\S])";this.BE={b:"\\\\.",r:0};this.ASM={cN:"string",b:"'",e:"'",i:"\\n",c:[this.BE],r:0};this.QSM={cN:"string",b:'"',e:'"',i:"\\n",c:[this.BE],r:0};this.CLCM={cN:"comment",b:"//",e:"$"};this.CBLCLM={cN:"comment",b:"/\\*",e:"\\*/"};this.HCM={cN:"comment",b:"#",e:"$"};this.NM={cN:"number",b:this.NR,r:0};this.CNM={cN:"number",b:this.CNR,r:0};this.BNM={cN:"number",b:this.BNR,r:0};this.inherit=function(r,s){var p={};for(var q in r){p[q]=r[q]}if(s){for(var q in s){p[q]=s[q]}}return p}}();hljs.LANGUAGES.cpp=function(){var a={keyword:{"false":1,"int":1,"float":1,"while":1,"private":1,"char":1,"catch":1,"export":1,virtual:1,operator:2,sizeof:2,dynamic_cast:2,typedef:2,const_cast:2,"const":1,struct:1,"for":1,static_cast:2,union:1,namespace:1,unsigned:1,"long":1,"throw":1,"volatile":2,"static":1,"protected":1,bool:1,template:1,mutable:1,"if":1,"public":1,friend:2,"do":1,"return":1,"goto":1,auto:1,"void":2,"enum":1,"else":1,"break":1,"new":1,extern:1,using:1,"true":1,"class":1,asm:1,"case":1,typeid:1,"short":1,reinterpret_cast:2,"default":1,"double":1,register:1,explicit:1,signed:1,typename:1,"try":1,"this":1,"switch":1,"continue":1,wchar_t:1,inline:1,"delete":1,alignof:1,char16_t:1,char32_t:1,constexpr:1,decltype:1,noexcept:1,nullptr:1,static_assert:1,thread_local:1,restrict:1,_Bool:1,complex:1},built_in:{std:1,string:1,cin:1,cout:1,cerr:1,clog:1,stringstream:1,istringstream:1,ostringstream:1,auto_ptr:1,deque:1,list:1,queue:1,stack:1,vector:1,map:1,set:1,bitset:1,multiset:1,multimap:1,unordered_set:1,unordered_map:1,unordered_multiset:1,unordered_multimap:1,array:1,shared_ptr:1}};return{dM:{k:a,i:"</",c:[hljs.CLCM,hljs.CBLCLM,hljs.QSM,{cN:"string",b:"'\\\\?.",e:"'",i:"."},{cN:"number",b:"\\b(\\d+(\\.\\d*)?|\\.\\d+)(u|U|l|L|ul|UL|f|F)"},hljs.CNM,{cN:"preprocessor",b:"#",e:"$"},{cN:"stl_container",b:"\\b(deque|list|queue|stack|vector|map|set|bitset|multiset|multimap|unordered_map|unordered_set|unordered_multiset|unordered_multimap|array)\\s*<",e:">",k:a,r:10,c:["self"]}]}}}();hljs.LANGUAGES.r={dM:{c:[hljs.HCM,{cN:"number",b:"\\b0[xX][0-9a-fA-F]+[Li]?\\b",e:hljs.IMMEDIATE_RE,r:0},{cN:"number",b:"\\b\\d+(?:[eE][+\\-]?\\d*)?L\\b",e:hljs.IMMEDIATE_RE,r:0},{cN:"number",b:"\\b\\d+\\.(?!\\d)(?:i\\b)?",e:hljs.IMMEDIATE_RE,r:1},{cN:"number",b:"\\b\\d+(?:\\.\\d*)?(?:[eE][+\\-]?\\d*)?i?\\b",e:hljs.IMMEDIATE_RE,r:0},{cN:"number",b:"\\.\\d+(?:[eE][+\\-]?\\d*)?i?\\b",e:hljs.IMMEDIATE_RE,r:1},{cN:"keyword",b:"(?:tryCatch|library|setGeneric|setGroupGeneric)\\b",e:hljs.IMMEDIATE_RE,r:10},{cN:"keyword",b:"\\.\\.\\.",e:hljs.IMMEDIATE_RE,r:10},{cN:"keyword",b:"\\.\\.\\d+(?![\\w.])",e:hljs.IMMEDIATE_RE,r:10},{cN:"keyword",b:"\\b(?:function)",e:hljs.IMMEDIATE_RE,r:2},{cN:"keyword",b:"(?:if|in|break|next|repeat|else|for|return|switch|while|try|stop|warning|require|attach|detach|source|setMethod|setClass)\\b",e:hljs.IMMEDIATE_RE,r:1},{cN:"literal",b:"(?:NA|NA_integer_|NA_real_|NA_character_|NA_complex_)\\b",e:hljs.IMMEDIATE_RE,r:10},{cN:"literal",b:"(?:NULL|TRUE|FALSE|T|F|Inf|NaN)\\b",e:hljs.IMMEDIATE_RE,r:1},{cN:"identifier",b:"[a-zA-Z.][a-zA-Z0-9._]*\\b",e:hljs.IMMEDIATE_RE,r:0},{cN:"operator",b:"<\\-(?!\\s*\\d)",e:hljs.IMMEDIATE_RE,r:2},{cN:"operator",b:"\\->|<\\-",e:hljs.IMMEDIATE_RE,r:1},{cN:"operator",b:"%%|~",e:hljs.IMMEDIATE_RE},{cN:"operator",b:">=|<=|==|!=|\\|\\||&&|=|\\+|\\-|\\*|/|\\^|>|<|!|&|\\||\\$|:",e:hljs.IMMEDIATE_RE,r:0},{cN:"operator",b:"%",e:"%",i:"\\n",r:1},{cN:"identifier",b:"`",e:"`",r:0},{cN:"string",b:'"',e:'"',c:[hljs.BE],r:0},{cN:"string",b:"'",e:"'",c:[hljs.BE],r:0},{cN:"paren",b:"[[({\\])}]",e:hljs.IMMEDIATE_RE,r:0}]}};
hljs.initHighlightingOnLoad();
</script>



<style type="text/css">
body, td {
   font-family: sans-serif;
   background-color: white;
   font-size: 13px;
}

body {
  max-width: 800px;
  margin: auto;
  padding: 1em;
  line-height: 20px;
}

tt, code, pre {
   font-family: 'DejaVu Sans Mono', 'Droid Sans Mono', 'Lucida Console', Consolas, Monaco, monospace;
}

h1 {
   font-size:2.2em;
}

h2 {
   font-size:1.8em;
}

h3 {
   font-size:1.4em;
}

h4 {
   font-size:1.0em;
}

h5 {
   font-size:0.9em;
}

h6 {
   font-size:0.8em;
}

a:visited {
   color: rgb(50%, 0%, 50%);
}

pre, img {
  max-width: 100%;
}
pre {
  overflow-x: auto;
}
pre code {
   display: block; padding: 0.5em;
}

code {
  font-size: 92%;
  border: 1px solid #ccc;
}

code[class] {
  background-color: #F8F8F8;
}

table, td, th {
  border: none;
}

blockquote {
   color:#666666;
   margin:0;
   padding-left: 1em;
   border-left: 0.5em #EEE solid;
}

hr {
   height: 0px;
   border-bottom: none;
   border-top-width: thin;
   border-top-style: dotted;
   border-top-color: #999999;
}

@media print {
   * {
      background: transparent !important;
      color: black !important;
      filter:none !important;
      -ms-filter: none !important;
   }

   body {
      font-size:12pt;
      max-width:100%;
   }

   a, a:visited {
      text-decoration: underline;
   }

   hr {
      visibility: hidden;
      page-break-before: always;
   }

   pre, blockquote {
      padding-right: 1em;
      page-break-inside: avoid;
   }

   tr, img {
      page-break-inside: avoid;
   }

   img {
      max-width: 100% !important;
   }

   @page :left {
      margin: 15mm 20mm 15mm 10mm;
   }

   @page :right {
      margin: 15mm 10mm 15mm 20mm;
   }

   p, h2, h3 {
      orphans: 3; widows: 3;
   }

   h2, h3 {
      page-break-after: avoid;
   }
}
</style>



</head>

<body>
<p>This vignette introduces how to use <em>SimPhe</em>, an R package dedicated to simulate multiple phenotyes based on genotyping data. The main feature of this package is the possibility to take genetic epistatic effects, not only additive-additive interaction but also additive-dominance and dominance-dominance, heritability and multiple phenotypes into account. Moreover, we provide a variety of convenient functions, including suggestion for set the variation of random variable according to user specified genetic effects and flexible support for input formats, as well as output formats. It also supports the input of plink formats.</p>

<h2>Installation</h2>

<pre><code class="r">install.packages(SimPhe)
</code></pre>

<h2>Sample implementation</h2>

<pre><code class="r"># load package
library(SimPhe)
</code></pre>

<p>First, we show how easy to get phenotype(s) by using <em>sim.phe</em>. Before running <em>sim.phe</em>, we need specify the parameter file and genotype file for simulation. After installing <em>SimPhe</em>, these two toy files exist in package folder. To get the path, type</p>

<pre><code class="r"># get file path of simulation parameters
# (two shared SNP pairs and one independent SNP pair for each phenotype)
fpar.path &lt;- system.file(&quot;extdata&quot;, &quot;simupars.txt&quot;, package=&quot;SimPhe&quot;)

# get file path of genotype file: rows are individuals and columns are SNPs
fgeno.path &lt;- system.file(&quot;extdata&quot;, &quot;10SNP.txt&quot;, package=&quot;SimPhe&quot;)
</code></pre>

<p>Then simulate the phenotypes as designed in the parameter file after loading package:</p>

<pre><code class="r"># simulate phenotype(s)
phe &lt;- sim.phe(fgenetic.pars = fpar.path, 
               fgeno = fgeno.path, 
               ftype = &quot;snp.head&quot;, 
               seed = 123, 
               fwrite = FALSE)
</code></pre>

<p>In the parameter file, we conduct two phenotypes contrbituted by two common SNP pairs with epistatic effects and one independent SNP pair with epistatic effects, the example of simulation parameters, same as the example file, is also included in <em>SimPhe</em>. Users could touch it by typing <em>genepars</em>.</p>

<pre><code class="r">genepars
</code></pre>

<pre><code>## $P1mean
##   mean
## 1   10
## 
## $P1main
##     SNP additive dominance
## 1 SNP01      4.0       2.0
## 2 SNP02      2.5       1.5
## 3 SNP03      5.5       1.8
## 4 SNP04     -2.5       1.0
## 5 SNP05      8.5       9.8
## 6 SNP06      8.7       8.4
## 
## $P1epistasis
##    SNPA  SNPB additive_additive additive_dominance dominance_additive
## 1 SNP01 SNP02              10.0                2.0                2.8
## 2 SNP03 SNP04               8.0                3.0                3.5
## 3 SNP05 SNP06              20.5               20.6               20.4
##   dominance_dominance
## 1                 1.8
## 2                 2.5
## 3                10.2
## 
## $P1heritability
##   heritability
## 1          0.6
## 
## $P2mean
##   mean
## 1   20
## 
## $P2main
##     SNP additive dominance
## 1 SNP01      8.0       2.0
## 2 SNP02      4.5       1.8
## 3 SNP03      6.5       1.5
## 4 SNP04     -3.5      -1.2
## 5 SNP07     10.4       8.6
## 6 SNP08     12.3      11.9
## 
## $P2epistasis
##    SNPA  SNPB additive_additive additive_dominance dominance_additive
## 1 SNP01 SNP02              15.0                4.0                3.0
## 2 SNP03 SNP04              12.0                5.0                3.5
## 3 SNP07 SNP08              21.5               20.9               20.5
##   dominance_dominance
## 1                 2.0
## 2                 2.5
## 3                20.3
</code></pre>

<p>Phenotype 1 has been set with certain heritability but phenotype 2 has not. We will show whether the heritability of the simulated phenotype 1 is same as the set. <em>SimPhe</em> includes the coefficients and the allele frequencies for simulating phenotype 1: <em>gene.coef</em> and <em>allele.freq</em>, which is extracted from the simulation parameters.</p>

<pre><code class="r"># regression coefficients for simulating phenotype 1
gene.coefficients
</code></pre>

<pre><code>## $epi.par1
##    SNPA  SNPB additiveA dominanceA additiveB dominanceB additive_additive
## 1 SNP01 SNP02         4          2       2.5        1.5                10
##   additive_dominance dominance_additive dominance_dominance
## 1                  2                2.8                 1.8
## 
## $epi.par2
##    SNPA  SNPB additiveA dominanceA additiveB dominanceB additive_additive
## 1 SNP03 SNP04       5.5        1.8      -2.5          1                 8
##   additive_dominance dominance_additive dominance_dominance
## 1                  3                3.5                 2.5
## 
## $epi.par3
##    SNPA  SNPB additiveA dominanceA additiveB dominanceB additive_additive
## 1 SNP05 SNP06       8.5        9.8       8.7        8.4              20.5
##   additive_dominance dominance_additive dominance_dominance
## 1               20.6               20.4                10.2
</code></pre>

<pre><code class="r"># allele frequencies of SNPs for simulating phenotype 1
allele.freq
</code></pre>

<pre><code>##     SNP major.frequency minor.frequency
## 1 SNP05       0.6447673       0.3552327
## 2 SNP01       0.7238366       0.2761634
## 3 SNP03       0.7394188       0.2605812
## 4 SNP06       0.7334176       0.2665824
## 5 SNP02       0.6433986       0.3566014
## 6 SNP04       0.6581386       0.3418614
</code></pre>

<pre><code class="r"># calculate genetic variance
genevar &lt;- calc.gene.var(gene.coefficients, allele.freq)

# the variance of simulated phenotype 1
phe1var &lt;- var(phe[, &quot;p1&quot;])

# heritability of simulated phenotype 1
simuht &lt;- genevar / phe1var
simuht
</code></pre>

<pre><code>## [1] 0.5793987
</code></pre>

<p>The result is not the exactly 0.6 due to the (pseudo)random numbers generated in R.
To get phenotype 2 with a specific heritability, for example, 0.45, we
could proceed as:</p>

<pre><code class="r"># get regression coefficients settings for phenotype 2
genecoef &lt;- get.gene.coef(main.pars = specify.pars(genetic.pars = genepars,
                                                   effect.type  = &quot;main&quot;,
                                                   phe.index    = 2),
                          epi.pars  = specify.pars(genetic.pars = genepars,
                                                   effect.type  = &quot;epistasis&quot;,
                                                   phe.index    = 2))
# get genotype data
genotype &lt;- read.geno(fname = fgeno.path, ftype = &quot;snp.head&quot;)

# get allele frequencies of the SNPs set for phenotype 2
freq2 &lt;- get.freq(geno = genotype,
                  epi.pars  = specify.pars(genetic.pars = genepars,
                                           effect.type  = &quot;epistasis&quot;,
                                           phe.index    = 2))

# get noise variation
exp.noise.var &lt;- get.noise.var(gene.coef = genecoef,
                               freq = freq2,
                               heritability = 0.45)
</code></pre>

<p>Then when simulating a phenotype, just give this value as argument
<em>noise.var</em> to function <em>sim.phe</em>. It will generate a
phenotype which has a heritability close to 0.45.</p>

<p>As we mentioned earlier in this article, buiding the correlation by
setting the shared interactive SNP pairs cannot be controlled. We can take a look at the correlation between the simulated phenotype 1 and phenotype 2:</p>

<pre><code class="r"># correlation test
cor.test(phe[, &quot;p1&quot;], phe[, &quot;p2&quot;])
</code></pre>

<pre><code>## 
##  Pearson&#39;s product-moment correlation
## 
## data:  phe[, &quot;p1&quot;] and phe[, &quot;p2&quot;]
## t = 18.111, df = 4747, p-value &lt; 2.2e-16
## alternative hypothesis: true correlation is not equal to 0
## 95 percent confidence interval:
##  0.227434 0.280645
## sample estimates:
##       cor 
## 0.2542319
</code></pre>

<p>According to the result of correlation test, the two simulated
phenotypes are significantly correlated but the correlation
coefficient is small and the amount cannot easily be predicted. To get
a certain amount or higher correlation, we can conduct correlation by
applying the correlation matrix to two independent variables. For two
phenotypes, if we set different SNP pairs for each, we assume these
two phenotypes are independent. Here we give another parameter file
to but same genotype file to the arguments in <em>sim.phe</em>,
<em>fgenetic.pars</em> and <em>fgeno</em>:</p>

<pre><code class="r"># get file path of simulation parameters
# (different SNP pairs for each phenotype)
fpar.path &lt;- system.file(&quot;extdata&quot;, &quot;sep_simupars.txt&quot;, package=&quot;SimPhe&quot;)

# simulate phenotype(s)
indphe &lt;- sim.phe(fgenetic.pars = fpar.path,
                  fgeno = fgeno.path,
                  ftype = &quot;snp.head&quot;,
                  seed = 123,
                  fwrite = FALSE)
</code></pre>

<p>We can test the correlation between initial phenotypes with seperated SNP pairs settings:</p>

<pre><code class="r"># test original correlation
cor.test(indphe[, &quot;p1&quot;], indphe[, &quot;p2&quot;])
</code></pre>

<pre><code>## 
##  Pearson&#39;s product-moment correlation
## 
## data:  indphe[, &quot;p1&quot;] and indphe[, &quot;p2&quot;]
## t = 1.2542, df = 4747, p-value = 0.2098
## alternative hypothesis: true correlation is not equal to 0
## 95 percent confidence interval:
##  -0.01024663  0.04661945
## sample estimates:
##        cor 
## 0.01820113
</code></pre>

<p>Apparently, these two phenotypes are not related. We could continue our work on coverting them to be correlated. First, conduct a correlation matrix:</p>

<pre><code class="r"># correlation matrix
corm &lt;- matrix(c(1, 0.6, 0.6, 1), ncol = 2)
corm
</code></pre>

<pre><code>##      [,1] [,2]
## [1,]  1.0  0.6
## [2,]  0.6  1.0
</code></pre>

<p>Before applying correlation matrix to simulated phenotypes, we would like to know what the data looks like:</p>

<pre><code class="r">apply(indphe, 2, mean)
</code></pre>

<pre><code>##       p1       p2 
## 20.22581 32.92018
</code></pre>

<pre><code class="r">apply(indphe, 2, sd)
</code></pre>

<pre><code>##       p1       p2 
## 21.45382 18.08358
</code></pre>

<p>Then we can build correlation between the two initial phenotypes:</p>

<pre><code class="r"># build correlation
corphe &lt;- build.cor.phe(indphe, corMtr = corm)
</code></pre>

<p>Now let&#39;s test the correlation between the two new phenotypes and see if there is anything difference:</p>

<pre><code class="r"># check mean and standard deviation of new data set
apply(corphe, 2, mean)
</code></pre>

<pre><code>##       p1       p2 
## 20.22581 32.92018
</code></pre>

<pre><code class="r">apply(corphe, 2, sd)
</code></pre>

<pre><code>##       p1       p2 
## 21.45382 18.24088
</code></pre>

<pre><code class="r"># test correlation
cor.test(corphe[, &quot;p1&quot;], corphe[, &quot;p2&quot;])
</code></pre>

<pre><code>## 
##  Pearson&#39;s product-moment correlation
## 
## data:  corphe[, &quot;p1&quot;] and corphe[, &quot;p2&quot;]
## t = 52.937, df = 4747, p-value &lt; 2.2e-16
## alternative hypothesis: true correlation is not equal to 0
## 95 percent confidence interval:
##  0.5910611 0.6268411
## sample estimates:
##       cor 
## 0.6092611
</code></pre>

<p>Obviously, there is no significant difference on means and standard deviations between the initial phenotypes and the new phenotypes.</p>

</body>

</html>
